server.js

```diff
@@ -7,6 +7,16 @@ var app = express();
 
 app.use(bodyParser.json());
 
+app.use(function maskIndex(req, res, next) {
+  // Adding this redirect to simplify caching a recipe page,
+  // essentially so we don't have to cache "/" and "/index.html"
+  // So: "recipe/index.html" -> "recipe/" , "index.html?123" -> "?123"
+  if (req.url.match(/\/(.*)\/index\.html\??(.*)$/)) {
```

Comment:
nit: use `test` and not `match`

An alternative might be using `url.parse` and applying the regexp (or `endsWidth`) on the `pathname` value.